lib/3scale/backend/validators.rb

```diff
@@ -21,6 +21,13 @@ module Validators
       OAUTH_VALIDATORS = ([Validators::OauthSetting,
                            Validators::OauthKey,
                            Validators::RedirectURI] + COMMON_VALIDATORS).freeze
+
+      # OIDC specific validators will only check app keys when app_key is given.
+      #
+      # No need to add OauthSetting, since we need to check that to tell
+      # OIDC apart from the rest when calling authrep.xml (note lack of
+      # the oauth_ prefix).
+      OIDC_VALIDATORS = ([Validators::OauthKey] + COMMON_VALIDATORS).freeze
```

Comment:
Related to https://github.com/3scale/apisonator/pull/280#discussion_r645386500

Reply:
Apisonator does not know anything about OIDC: notice that OIDC does not have any additional information like referrer uris in this phase nor does it (it should not) check any secrets - it all happens before 3scale sees it. With OAuth (and more so when using the removed token infrastructure) you had some workflows in which you needed checks at this level (ie. 3scale was acting as a poor-man's identity provider and verifier). With OIDC 3scale cannot do that, and it does not get to see any extra details like whatever is encoded in the JWTs or other administrative settings in the application, which belong in the IdP.

So the `OauthKey` validator above is not really needed, it should be gone, but it is kept so that the current behavior regarding the checks when calling authorize and authrep is maintained. That is, the change is limited to avoid checking application keys and not removing questionable previous behavior to avoid potentially breaking (wrong) usages.

Another thing is removing the OAuth specific validators that no longer make sense if 3scale is only handling user_key (btw, the docs say user_key is deprecated...), app_id/app_key and OIDC (app_id).

Reply:
All of what you said is true, what I meant is that now, OIDC is the only oauth flow supported, so having OIDC_VALIDATORS and OAUTH_VALIDATORS is confusing.

Reply:
Yes, but then we still might want to not break anything - do you know whether there is anyone still using this old/wrong behavior?

Reply:
Right, I forgot about that `RedirectURI` validator :thinking: 
I don't know if that's still used. It should not, but we should check carefully before getting rid of it.